packages/nimble-components/src/table/index.ts

```diff
@@ -37,16 +46,35 @@ export class Table<
     @observable
     public columnHeaders: string[] = [];
 
+    public get validity(): TableValidity {
+        return {
+            duplicateRowId: this.duplicateRowId,
+            missingRowId: this.missingRowId,
+            invalidRowId: this.invalidRowId
+        };
+    }
+
     private readonly table: TanStackTable<TData>;
     private options: TanStackTableOptionsResolved<TData>;
     private readonly tableInitialized: boolean = false;
 
+    private duplicateRowId = false;
+    private missingRowId = false;
+    private invalidRowId = false;
```

Comment:
thought on breaking this off into a separate class like `TableValidator` in a spot like `table/models/table-validator` to capture the different properties and generate the TableValidity object? Seems like it could be a pretty self contained class with methods like reset(), validateData(), getTableValidity(), and isValid(), etc. Could also split out its unit tests.

Reply:
Compared to other controls I'm more concerned about the table class getting really big

Reply:
I've added a `TableValidator` class with a bunch of tests. I removed some of the tests from the table component itself, but I kept a few so that we could validate that the validity state bubbles up correctly through the component and that the component renders as expected based on whether or not the ID is valid.